dafni/src/main_dafni.py

```diff
@@ -91,53 +84,33 @@ def get_args(test_args=None) -> argparse.Namespace:
     return args
 
 
-def read_variables(variables_path: Path) -> FileNotFoundError | dict:
+def parse_variables(causal_dag: CausalDAG) -> tuple:
     """
-    Function to read the variables.json file specified by the user
-    :param variables_path: A Path object of the user-specified file path
+    Function to parse and validate the variables defined in the causal dag.
+    :param causal_dag: an instancce of the CausalDAG class containing the relationships and variables (as attribtutes)
     :returns:
-            - dict - A valid dictionary consisting of the causal tests
+    - Tuple containing the inputs, outputs and constraints to pass into the modelling scenario
     """
-    if not variables_path.exists() or variables_path.is_dir():
-        print(f"JSON file not found at the specified location: {variables_path}")
 
-        raise FileNotFoundError
+    inputs = [
```

Comment:
Since we just return the inputs and outputs as separate lists and then join them up later, would it make sense to just return one list of `variables` defined as
```
variables = [
        Input(node, eval(eval(attributes["datatype"]))) if attributes["typestring"] == input else
        Output(node, eval(eval(attributes["datatype"]))) if attributes["typestring"] == input
        for node, attributes in causal_dag.graph.nodes(data=True)
]
```
It's not really a problem either way, but it's just a bit more elegant

Reply:
@jmafoster1 I think it's less clear here what's being calculated and returned as a single line. You'd then have to define `inputs = variables[0]` which is less informative than the current method.

Reply:
My point was that we never use `inputs` or `outputs` in isolation. We build them separately and return them separately, but it looks like the only thing we do with them is append them anyway, so I was wondering whether it made sense to treat them separately at all, but if you think it's more elegant the way you have it then we can leave it as is. It's not like people will be dealing with millions of inputs/outputs.